From: Mike Hommey <glandium@debian.org>
Date: Sat, 21 Jun 2008 03:09:21 +0200
Subject: [PATCH] Add another preferences directory for applications: preferences/syspref

This can be used to symlink to /etc/ice*/pref.
---
 mozilla/toolkit/xre/nsXREDirProvider.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/mozilla/toolkit/xre/nsXREDirProvider.cpp b/mozilla/toolkit/xre/nsXREDirProvider.cpp
index fd9411f..8f64aa4 100644
--- a/mozilla/toolkit/xre/nsXREDirProvider.cpp
+++ b/mozilla/toolkit/xre/nsXREDirProvider.cpp
@@ -624,6 +624,7 @@ nsXREDirProvider::LoadAppBundleDirs()
 }
 
 static const char *const kAppendPrefDir[] = { "defaults", "preferences", nsnull };
+static const char *const kAppendSysPrefDir[] = { "defaults", "syspref", nsnull };
 
 #ifdef DEBUG_bsmedberg
 static void
@@ -680,6 +681,7 @@ nsXREDirProvider::GetFilesInternal(const char* aProperty,
     LoadAppDirIntoArray(mXULAppDir, kAppendPrefDir, directories);
     LoadDirsIntoArray(mAppBundleDirectories,
                       kAppendPrefDir, directories);
+    LoadAppDirIntoArray(mXULAppDir, kAppendSysPrefDir, directories);
 
     rv = NS_NewArrayEnumerator(aResult, directories);
   }
-- 
